components/proxy/src/main/java/com/hotels/styx/infrastructure/FileBackedRegistry.java

```diff
@@ -122,11 +131,12 @@ public String fileName() {
     }
 
     private boolean updateResources(byte[] content, HashCode hashCode) {
-
-        Iterable<T> resources = reader.read(content);
+        Collection<T> resources = reader.read(content);
         Changes<T> changes = changes(resources, get());
 
         if (!changes.isEmpty()) {
+            checkState(resourceConstraint.test(resources));
```

Comment:
IMO this check should take place in the abstract registry `set` method. That way the check is not limited to the file based providers only.

Reply:
Also, it is not checking for resource constrains, such as limits in RAM, disk or CPU. It is checking that the configuration is internally consistent

So perhaps this check needs to be renamed.

Reply:
The name `resourceConstraint` is consistent with the variable `resources` that it is checking. What would you name the objects that registries contain?

I will look into moving the functionality to `AbstractRegistry.set`.

Reply:
Ok I see.  Regardless, thinking from `AbstractRegistry` point of view, it doesn't have a concept of resources "resources", just "registry". Perhaps call it a `registryConstraint`?

Reply:
`AbstractRegistry` has a method ` protected static <T extends Identifiable> Changes<T> changes(Iterable<T> newResources, Iterable<T> currentResources)`, so the term is already used. We can still rename them to something else though.
`registryConstraint` is not really informative though, that would be naming it after the object it lives in, rather than the objects it constrains.
If we rename "resources" throughout the code, I will rename the constraint accordingly.